### DISCUSSION TITLE
Updating from a previous release

### DISCUSSION THREAD

**Discussion by ferrellsl** [Joshua-Riek/ubuntu-rockchip#34] [General]
What's the best way to upgrade from release 1.6 to 1.7 without having to reinstall all my apps and settings?  Is there a way to do a distro upgrade?

**Comment by Joshua-Riek** [comment]
Sadly the only way to upgrade from a previous release is with a fresh install. I want to create a script to upgrade automatically, but for it to work correctly would take a while. I'm still focused on bug fixing, optimizations, and adding new/missing features.

**Reply by ypatriot** [reply]
helo, 
I'm doing this to solve the problem:
Backup:
dpkg --get-selections > ~/Package.list
sudo cp -R /etc/apt/sources.list* ~/
sudo apt-key exportall > ~/Repo.keys
copy of /var/cache/apt/archives/*deb to another storage, 
//-------------------------------------------------------------------------------
Restore:
copy these debs package to new installed system folder:  var/cache/apt/archives
sudo apt-key add ~/Repo.keys
sudo cp -R ~/sources.list* /etc/apt/
sudo apt-get update
sudo apt-get install dselect
sudo dselect update
sudo dpkg --set-selections < ~/Package.list
sudo apt-get dselect-upgrade -y

this is the reference:
https://gist.github.com/reloxx13/81d8bc8e37b5ff1aa970e1adc9da7955
